marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/entitySearch/opticPlanExportService.sjs

```diff
@@ -0,0 +1,85 @@
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+'use strict';
+
+const op = require('/MarkLogic/optic');
+const search = require('/MarkLogic/appservices/search/search');
+const options = require("/data-hub/5/impl/hub-utils/invoke-search-options.sjs");
+
+const returnFlags = '<return-aggregates xmlns="http://marklogic.com/appservices/search">false</return-aggregates>\n' +
+  '<return-constraints xmlns="http://marklogic.com/appservices/search">false</return-constraints>\n' +
+  '<return-facets xmlns="http://marklogic.com/appservices/search">false</return-facets>\n' +
+  '<return-frequencies xmlns="http://marklogic.com/appservices/search">false</return-frequencies>\n' +
+  '<return-metrics xmlns="http://marklogic.com/appservices/search">false</return-metrics>\n' +
+  '<return-plan xmlns="http://marklogic.com/appservices/search">false</return-plan>\n' +
+  '<return-qtext xmlns="http://marklogic.com/appservices/search">false</return-qtext>\n' +
+  '<return-results xmlns="http://marklogic.com/appservices/search">false</return-results>\n' +
+  '<return-similar xmlns="http://marklogic.com/appservices/search">false</return-similar>\n' +
+  '<return-values xmlns="http://marklogic.com/appservices/search">false</return-values>\n' +
+  '<return-query xmlns="http://marklogic.com/appservices/search">true</return-query>\n';
+
+const stylesheet = fn.head(xdmp.unquote('<xsl:stylesheet version="1.0"\n' +
+  ' xmlns:xsl="http://www.w3.org/1999/XSL/Transform">\n' +
+  '\n' +
+  ' <xsl:template match="node()|@*">\n' +
+  '  <xsl:copy>\n' +
+  '   <xsl:apply-templates select="node()|@*"/>\n' +
+  '  </xsl:copy>\n' +
+  ' </xsl:template>\n' +
+  '\n' +
+  ' <xsl:template match="*:return-aggregates|*:return-constraints|*:return-facets|*:return-frequencies|*:return-metrics|*:return-plan|*:return-qtext|*:return-results|*:return-similar|*:return-values|*:return-query" />\n' +
+  '\n' +
+  ' <xsl:template match="*:options">\n' +
+  '  <xsl:copy>\n' +
+  '   <xsl:apply-templates select="node()|@*" />\n' +
+  returnFlags +
+  '  </xsl:copy>\n' +
+  ' </xsl:template>\n' +
+  '\n' +
+  '</xsl:stylesheet>'));
+
+const entityOptionsFileName = '/Default/data-hub-FINAL/rest-api/options/exp-final-entity-options.xml';
+
+var viewName;
+var schemaName;
+var limit;
+var structQuery;
+var searchQText;
+var columns = xdmp.getRequestField("columns");
+
+structQuery = fn.head(xdmp.unquote(structQuery)).root;
+searchQText = searchQText || "";
+
+const oldOptions = options.getSearchOptions(entityOptionsFileName);
```

Comment:
Given that we need an amp plus a separate module, which then has to invoke a function in the modules database for a document whose URI will be dependent on the groups - I think we should just generate these on the fly. Check out hub-entities.xqy.

Reply:
My understanding was that a user is free to edit the search options file.
Do we still allow a user to edit the options file for HC? 

Another alternative here could be to perform the search operation from the java middle-tier. Though it would increase a request, it would make sure that we use the correct search options file.

Reply:
If the user does edit the options file, it'll get overwritten when an entity model is modified. 

I like the idea of having the middle tier fetch the options and then pass them to the DS endpoint. That way, the endpoint doesn't have to worry about what group to use when finding the options.